File: dtc.frontend/src/dtos/Match/CreateMatchDto.ts
import z from "zod";
import { MatchStatus } from "../../enums/MatchStatus";

export const createMatchDtoSchema = z.object({
  roundId: z.number({
    message: "RoundId ist ein Pflichtfeld.",
  }),

  groupId: z.number().nullable(),

  boardId: z.number().nullable(),

  status: z.enum(MatchStatus).default(MatchStatus.Scheduled),
  plannedStart: z.iso.datetime({ offset: true }).nullable(),
  plannedEnd: z.iso.datetime({ offset: true }).nullable(),
});

export type CreateMatchDto = z.infer<typeof createMatchDtoSchema>;

File: DTC.Api/Dtos/Match/CreateMatchDto.cs
using DTC.Api.Enums;
using System.ComponentModel.DataAnnotations;

namespace DTC.Api.Dtos.Match
{
    public class CreateMatchDto
    {
        [Range(1, int.MaxValue, ErrorMessage = "Die Round ist ein Pflichtfeld.")]
        public int RoundId { get; set; }

        // Nur bei Gruppenspielen gesetzt
        public int? GroupId { get; set; }

        // Board ist optional
        public int? BoardId { get; set; }

        public MatchStatus Status { get; set; } = MatchStatus.Scheduled;

        public DateTimeOffset? PlannedStart { get; set; }

        public DateTimeOffset? PlannedEnd { get; set; }

        public DateTimeOffset? ActualStart { get; set; }

        public DateTimeOffset? ActualEnd { get; set; }
    }

}
